src/lib/Microsoft.Health.Events/Telemetry/Exceptions/EventHubConfigurationExceptionTelemetryProcessor.cs

```diff
@@ -0,0 +1,63 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using EnsureThat;
+using Microsoft.Health.Common.Telemetry;
+using Microsoft.Health.Logging.Telemetry;
+
+namespace Microsoft.Health.Events.Telemetry.Exceptions
+{
+    public class EventHubConfigurationExceptionTelemetryProcessor
+    {
+        private readonly HashSet<Type> _handledExceptions;
+
+        public EventHubConfigurationExceptionTelemetryProcessor()
+            : this(
+                typeof(InvalidEventHubException),
```

Comment:
I think having such an interface makes sense. 

The side note to this that I want to point out about "handled" exceptions is that there could be two interpretations with these being introduced now - 
1. Set of exceptions that we will expect during the processing of an event batch, that we will just catch, log and then continue the processing of next events and not stall . Refer usage of https://github.com/microsoft/iomt-fhir/blob/master/src/lib/Microsoft.Health.Fhir.Ingest/Telemetry/ExceptionTelemetryProcessor.cs 
In the above cases "handled" means we will not throw on these or stop processing.
2. Set of exceptions that we will expect during the processing of an event batch, that we will catch, convert to custom exceptions and log these custom exceptions, but unlike # 1 we will throw as we cannot just ignore these as this will keep occuring until the root cause is fixed, so we cannot move on to the next events.

Just calling out the above in case it influences how we want to model the IHandleable interface.

Reply:
That makes sense and we should consider it when making the interface.
For now I'll defer this change to when we implement @pallar-ms's later change that will better process handled vs. unhandled exceptions.

Reply:
Agreed. Perhaps two types of interfaces would be needed here: `IHandledAndContinue` and `IHandledAndThrow`. Or a flag which indicates if you can continue or need to throw. Something to that effect.